src/coreComponents/dataRepository/xmlWrapper.hpp

```diff
@@ -46,21 +46,220 @@ namespace geos
 
+class xmlDocument;
+
+/// Stores the source file path, and position in the file of a xml attribute.
+struct xmlAttributePos
+{
+  /// Path of the file containing this element
+  string filePath;
+  /// Line where the element is defined. Start at 1.
+  /// Default value is xmlDocument::npos
+  size_t const line;
+  /// Character offset of this element in the line that contains it (starting from 1)
+  /// Equals to xmlDocument::npos if it couldn't be determined.
+  size_t const offsetInLine;
+  /// Character offset of this element in the file that contains it (starting from 0)
+  /// Equals to xmlDocument::npos if it couldn't be determined.
+  size_t const offset;
+
+  /**
+   * @brief Constructor of this struct.
+   * @param filePath the path of the original xml file containing this attribute
+   * @param line Line where the element is defined. Start at 1.
+   * @param offsetInLine Character offset of this element in the line that contains it (starting from 1)
+   * @param offset Character offset of this element in the file that contains it (starting from 0)
+   */
+  xmlAttributePos( string const & filePath, size_t line, size_t offsetInLine, size_t offset );
+  /**
+   * @return false if the position is undefined.
+   */
+  bool isFound() const;
+  /**
+   * @return a string containing the file name and position in the file.
+   */
+  string toString() const;
+};
+
+/// Used to retrieve the position of dataRepository::Wrapper in XML
+struct xmlNodePos : xmlAttributePos
+{
+  /// Reference of the main xmlDocument that contains all original file buffers.
+  xmlDocument const & document;
+
+  /**
+   * @brief Constructor of this struct.
+   * @param document an xml document containing this node, or including a file which includes it
+   * @param filePath the path of the original xml file containing this node
+   * @param line Line where the node is defined. Start at 1.
+   * @param offsetInLine Character offset of this node in the line that contains it (starting from 1)
+   * @param offset Character offset of this node in the file that contains it (starting from 0)
+   */
+  xmlNodePos( xmlDocument const & document, string const & filePath, size_t line, size_t offsetInLine, size_t offset );
+  /**
+   * @return false if the position is undefined.
+   */
+  bool isFound() const;
+  /**
+   * @brief Compute the xmlAttributePos of an xml attribute
+   * @param attName the name of the attribute to locate
+   * @return an xmlAttributePos object that represents the position of the target node.
+   */
+  xmlAttributePos getAttributeLine( string const & attName ) const;
+};
+
+/// Wrapper class for the type of xml document.
+/// This class exists to intercept file / string loading methods, and to keep the loaded buffers,
+/// in order to retrieve the source file and line of nodes and attributes.
+class xmlDocument : public pugi::xml_document
+{
+public:
+  /// Error value for when an offset / line position is undefined.
+  static const size_t npos;
```

Comment:
So we have:
```cpp
  static constexpr size_t npos = string::npos;
```
... in the `.hpp`, and: 
```cpp
constexpr size_t xmlDocument::npos;
```
... in the `.cpp` (we need to keep this second one because of external use).

Reply:
Yeah you're right